pkg/imgpkg/imagedesc/types.go

```diff
@@ -133,3 +140,23 @@ func (t ImageOrIndex) Tag() string {
 		panic("Unknown item")
 	}
 }
+
+func (t ImageOrIndex) MountableImage(registry RegistryRemoteImage) (regv1.Image, error) {
+	if t.Image == nil {
+		return nil, fmt.Errorf("Unable to retrieve a mountable image on an imageindex")
+	}
+
+	reference, err := regname.ParseReference((*t.Image).Ref())
+	if err != nil {
+		return nil, err
+	}
+	descriptor, err := registry.Get(reference)
```

Comment:
if we are going to grab images from a remote source, we need to have a gurantee that reference points to an immutable image (digest). otherwise it could be a tag and this could return a random/mutable image.

Reply:
@cppforlife done